aioshelly/wsrpc.py

```diff
@@ -79,7 +101,11 @@ def _next_id(self) -> int:
         self._call_id += 1
         return self._call_id
 
-    async def connect(self, aiohttp_session: aiohttp.ClientSession) -> None:
+    async def connect(
+        self,
+        aiohttp_session: aiohttp.ClientSession,
+        options: ConnectionOptions | None = None,
```

Comment:
`options` here contains the `ip_address` already passed in the constructor. It isn't obvious that it is being ignored in the `connect()` method, along with the other properties of the `ConnectionOptions` instance.

If it were me, I'd simply pass an optional `password` in the constructor, and complicate it further if needed in the future.